core/src/main/scala/com/criteo/cuttle/Workflow.scala

```diff
@@ -90,6 +202,18 @@ trait Workflow[S <: Scheduling] {
       val edges = leftWorkflow.edges ++ rightWorkflow.edges ++ newEdges
     }
   }
+
+  case class DependencyBuilder(workflow: Workflow[S], parentJob: Job[S])(implicit dependencyDescriptor: S#DependencyDescriptor) {
+    def to(childJob: Job[S]): Workflow[S] = {
+      new Workflow[S] {
+        val vertices = workflow.vertices ++ Set(childJob, parentJob)
+        val edges = workflow.edges + ((childJob, parentJob, dependencyDescriptor))
+      }
+    }
+  }
+
+  def withDependencyFrom(parentJob: Job[S])(implicit dependencyDescriptor: S#DependencyDescriptor): DependencyBuilder =
```

Comment:
not sure it's worth it to add new API method just for tests. I think it's an important design decision to discuss before doing it

Reply:
Agreed. Will remove it.